coils_tet_mesh.py

```diff
@@ -0,0 +1,277 @@
+    function_names = split_list
+    return function_names
+
+
+def coils_tet_mesh(coils_path, total_coil_count, log_general_path,
+                   exo_path, h5m_path, csv_path):
+    """Create a tetrahedral mesh of the magnet coils using Cubit 
+    and export them as .exo and .h5m files
+
+        Arguments:
+            coils_path (str): absolute path to the coils .step file
+                containing all of the coil geometries to be meshed.
+            total_coil_count (int): number of coils contained in the .step file
+            log_general_path (str): absolute path to a .log file that will be
+                created by Cubit. 
+                **Note: Do not include any numerical index at the end of
+                the file name because the function iterates over every coil in the .step
+                file and will append the index to the .log filename.**
+            exo_path (str): absolute path to a .exo file that will be
+                created by the function. 
+            h5m_path (str): absolute path to a .h5m file that will be
+                created by the function. 
+            csv_path (str): absolute path to a .csv file that will store the top 
+                quality metrics data for each coil.
+        """
+
+    # GEOMETRY AND MESH CREATION
+
+    # Start Cubit - this step is key
+    cubit.init(['cubit', '-nojournal'])
+
+    # Import coils
+    cubit.cmd(f'import step "{coils_path}" heal')
+    cubit.cmd('volume  size auto factor 5')
+
+    # Split log path
+    log_split = log_general_path.split('.')
+    log_prefix = log_split[0]
+    log_ext = log_split[1]
+
+    # Define mesh quality metrics
+    quality_metric_list = ['shape', 'aspect ratio', 'condition no.',
+                           'distortion', 'element volume','equiangle skew', 'equivolume skew',
+                           'inradius', 'jacobian', 'normalized inradius', 'relative size',
+                           'scaled jacobian', 'shape and size']
+    
+    # Create an empty Pandas data frame to store the top metric results
+    top_results_cols = ['Function Name', 'Average','Standard Deviation',
+                                        'Minimum', 'Maximum']
+    top_results = pd.DataFrame(columns = top_results_cols)
+
+    # For-loop iterates over all of the coils included in the coils.step file
+    for coil_num in np.arange(1,total_coil_count + 1):
+        
+        # Create the mesh
+        cubit.cmd(f'volume {coil_num} scheme tetmesh')
+        cubit.cmd(f'mesh volume {coil_num}')
+
+        # Set log file path and initiate logging
+        log_path = f"{log_prefix}_{coil_num}.{log_ext}"
+        cubit.cmd(f'logging on file "{log_path}"')
+
+        # Run mesh quality metrics for each relevant metric
+        for metric in quality_metric_list:
+            qual_command = f'quality volume {coil_num} {metric} global draw mesh list'
+            mesh_analytics = cubit.cmd(qual_command)
```

Comment:
Separation of concerns & cognitive load:
Make this a function that creates and does the quality measures on each mesh